services/database/scripts.mjs

```diff
@@ -0,0 +1,329 @@
+          `INSERT INTO sourcify_sync (chain_id, address, match_type, created_at) VALUES ($1, $2, $3, $4) ON CONFLICT (chain_id, address) DO NOTHING`,
+          [
+            contract.chainId,
+            contract.address,
+            contract.matchType,
+            contract.timestamp,
+          ]
+        );
+        console.log(
+          `Contract inserted to the sourcify_sync table: \n ${[
+            contract.chainId,
+            contract.address,
+            contract.matchType,
+            contract.timestamp,
+          ]}`
+        );
+      }
+    } catch (e) {
+      console.error("Error while storing contract in the database");
+      console.error(e);
+      process.exit(3);
+    }
+
+    console.log("successfuly imported contracts from" + repositoryV1Path);
+    databasePool.end();
+  });
+
+program
+  .command("sync")
+  .description("Verifies contracts in the sourcify_sync table")
+  .argument(
+    "<string>",
+    "Sourcify instance url: e.g. https://sourcify.dev/server"
+  )
+  .argument(
+    "<string>",
+    "Path to repository v1 contracts folder (e.g. /Users/user/sourcify/repository/contracts)"
+  )
+  .option(
+    "-c, --chains [items]",
+    "List of chains separated by comma (e.g. 1,5,...)"
+  )
+  .option("-sf, --start-from [number]", "Start from a specific timestamp (ms)")
+  .option("-l, --limit [number]", "Limit of concurrent verifications (ms)")
+  .action(async (sourcifyInstance, repositoryV1Path, options) => {
+    if (
+      repositoryV1Path.split("/")[repositoryV1Path.length - 1] !== "contracts"
+    ) {
+      console.error(
+        "Passed repositoryV1 path is not correct: " + repositoryV1Path
+      );
+      process.exit(4);
+    }
+
+    const databasePool = new Pool({
+      host: process.env.POSTGRES_HOST,
+      port: process.env.POSTGRES_PORT,
+      database: process.env.POSTGRES_DB,
+      user: process.env.POSTGRES_USER,
+      password: process.env.POSTGRES_PASSWORD,
+      max: 5,
+    });
+
+    let activePromises = 0;
+    let limit = options.limit ? options.limit : 1;
+
+    let processedContracts = 0;
+    while (true) {
+      // Fetch next contract
+      let nextContract = await fetchNextContract(databasePool, options);
+      if (!nextContract) break; // Exit loop if no more contracts
+      options.startFrom = new Date(nextContract.created_at).getTime();
+
+      // Process contract if within activePromises limit
+      if (activePromises < limit) {
+        activePromises++;
+        processContract(
+          sourcifyInstance,
+          repositoryV1Path,
+          databasePool,
+          nextContract
+        )
+          .then((res) => {
+            if (res[0]) {
+              console.log(`Successfully sync: ${[res[1]]}`);
+            } else {
+              console.error(`Failed to sync ${[res[1]]}`);
+            }
+            activePromises--;
+            processedContracts++;
+          })
+          .catch((e) => {
+            console.error(e);
```

Comment:
Shouldn't it `activePromises--`?

Reply:
Also would be nice if we log the chain and address here. We can then look what went wrong

Reply:
> Shouldn't it activePromises--?

it's the line above, no?